jobsdb/jobsdb.go

```diff
@@ -3031,7 +3062,30 @@ func (jd *HandleT) updateJobStatusDSInTx(ctx context.Context, tx *sql.Tx, ds dat
 	return
 }
 
-/**
+func (jd *HandleT) createExceptionFunction() error {
+	sqlStatement := fmt.Sprintf(`
+	CREATE FUNCTION %s  	
+	RETURNS trigger
+	AS
+	$$
+	BEGIN RAISE EXCEPTION '%s' USING ERRCODE = '%s';
+	END;
+	$$
+	LANGUAGE plpgsql;`, fmt.Sprintf("%s()", lockTableInsertExceptionFuncName), lockTableInsertExceptionMsg, lockTableInsertExceptionCode)
+
+	_, err := jd.dbHandle.Exec(sqlStatement)
+
+	var e *pq.Error
+	if err != nil && errors.As(err, &e) {
+		if e.Code == postgresDuplicateFuncErr {
+			return nil
+		}
+	}
+	return err
+}
```

Comment:
"`CREATE OR REPLACE FUNCTION` will either create a new function, or replace an existing definition. To replace the current definition of an existing function, use `CREATE OR REPLACE  FUNCTION`."
ref: https://www.postgresql.org/docs/current/sql-createfunction.html

Since, create or replace replaces the existing func, do you think it is a good idea to unnecessarily overwrite the func again and again?

Reply:
Either ways, I guess we should move this to startup to avoid doing this on every DS creation (with CREATE / CREATE or REPLACE) functions